nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/handler/AbstractBatchHandler.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * (C) Copyright 2018 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Luís Duarte <lduarte@nuxeo.com>
+ *     Florent Guillaume <fguillaume@nuxeo.com>
+ *
+ */
+package org.nuxeo.ecm.automation.server.jaxrs.batch.handler;
+
+import static org.nuxeo.ecm.automation.server.jaxrs.batch.BatchManagerComponent.CLIENT_BATCH_ID_FLAG;
+
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.Batch;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.BatchHandler;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.transientstore.api.TransientStore;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+
+public abstract class AbstractBatchHandler implements BatchHandler {
+    private static final Log log = LogFactory.getLog(AbstractBatchHandler.class);
+
+    private String name;
+
+    protected AbstractBatchHandler(String name) {
+        this();
+        this.name = name;
+    }
+
+    protected AbstractBatchHandler() {
+
+    }
+
+    @Override public String getName() {
+        return name;
+    }
+
+    @Override public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override public abstract Batch newBatch();
+
+    @Override public abstract Batch getBatch(String batchId);
+
+    @Override public abstract Batch newBatch(String providedId);
```

Comment:
These 3 methods are defined in the interface and could be removed, according to the [Abstract methods should not be redundant](https://sonarcloud.io/organizations/nuxeo/rules?open=squid%3AS3038&q=S3038) Sonar rule.